ls1intum/Artemis
# Update C simulation to new template

### Checklist

### Motivation and Context
The C exercise template was changed in https://github.com/ls1intum/Artemis/pull/1457, but the changes were not reflected to the K6 simulation that tests using the template solution.

This PR updates the simulation.

### Description
The sample solution is adjusted to be the one from the template. Sadly, some tests had to be removed from the template as they were flaky, therefore the build now always either fails (if a compile error occurs) or passes (if it compiles).
I've added a workaround to reflect this behaviour which should be removed once the template is adjusted (test cases are added that can partly fail or pass).

### Steps for Testing
I've enabled the server API tests for this branch: https://bamboo.ase.in.tum.de/browse/ARTEMIS-SAT0.
You can verify it is working by running the plan (make sure to block test server 3 for this as a deployment is performed).
